### PR TITLE
doc: Fix 'pbkdf2Sync' encoding problem

### PR DESCRIPTION
**Pull request by ghost**
Ref: https://github.com/nodejs/nodejs.org/pull/1796.

Since the '/newUser' request is encoded by 'sha512', and your
corresponding '/auth' must also use 'sha512' as the digest encoding
as well. Otherwise your decoding isn't right.

### CODE REVIEW

**Review by lpinca — APPROVED** [review]

**Comment by vsemozhetbyt** [comment]
cc @nodejs/crypto

**Comment by trevorah** [comment]
😳 whoops, sorry I didnt notice that in my original PR. Good spot @Maledong!

File: locale/en/docs/guides/simple-profiling.md [hunk 1]
```diff
@@ -63,7 +63,8 @@ app.get('/auth', (req, res) => {
     return res.sendStatus(400);
   }
 
-  const hash = crypto.pbkdf2Sync(password, users[username].salt, 10000, 512);
+  const salt = users[username].salt;
```
Comment (tniessen) [inline locale/en/docs/guides/simple-profiling.md]: Nit: Use destructuring.

**Review by tniessen — COMMENTED** [review]
Unrelated: 512 is an unusual key size (note that the key size parameter is in bytes, not bits). This might have originated from the misconception that it must match the output size of the hash function?

**Comment by trevorah** [comment]
@tniessen I picked sha512 for this guide because its used in the crypto examples (https://nodejs.org/api/crypto.html#crypto_crypto_pbkdf2_password_salt_iterations_keylen_digest_callback) which this guide appears to mirror.

**Comment by tniessen** [comment]
@trevorah SHA512 is a good choice, just the output key size is unusual (512 bytes!). This works equally well:

```js
crypto.pbkdf2Sync('password', 'salt', 10000, 32, 'sha512')
```

PBKDF2 is usually used to derive symmetric keys or to store passwords, neither of these applications requires keys with a size of 512 bytes.

**Comment by trevorah** [comment]
@tniessen hmm... weird. The only `crypto.pbkdf2*` examples that use a 512 key size are the ones that set `crypto.DEFAULT_ENCODING = 'hex'`. All the others use a 64 byte key size. Maybe the original author wanted to show some difference when setting `DEFAULT_ENCODING`?

**Comment by tniessen** [comment]
Uhh `DEFAULT_ENCODING`... What a design mistake. It still doesn't really matter, if you set the `DEFAULT_ENCODING` to `hex`, the output will still be a 512-byte (1024 hex digits) sequence, just encoded. But again, it is unrelated, so doesn't need to be changed in this PR. Also, it's not wrong, just unusual.

File: locale/en/docs/guides/simple-profiling.md (hunk 1 above)
Comment (ghost) [reply locale/en/docs/guides/simple-profiling.md]: Sorry for mistake, and I'll fix that sooooooooon :)

**Comment by ghost** [comment]
@trevorah：No problem. Just found that by accident when reading lines :)

File: locale/en/docs/guides/simple-profiling.md [hunk 2]
```diff
@@ -63,9 +63,10 @@ app.get('/auth', (req, res) => {
     return res.sendStatus(400);
   }
 
-  const hash = crypto.pbkdf2Sync(password, users[username].salt, 10000, 512);
+  const { salt, hash } = users[username];
+  const encryptHash = crypto.pbkdf2Sync(password, salt, 10000, 512, 'sha512');
 
-  if (users[username].hash.toString() === hash.toString()) {
+  if (hash.toString() === encryptHash.toString()) {
```
Comment (tniessen) [inline locale/en/docs/guides/simple-profiling.md]: Also unrelated: if `hash` and `encryptHash` are buffers, the best way to check for equality is `hash.equals(encryptHash)`.

Comment (LinusU) [reply locale/en/docs/guides/simple-profiling.md]: This should probably be a constant time comparison though? As not to leak the expected hash? (`crypto.timingSafeEqual`)

Comment (tniessen) [reply locale/en/docs/guides/simple-profiling.md]: Oh, right, good point, didn't look at the code as a whole to be honest.

Comment (ghost) [reply locale/en/docs/guides/simple-profiling.md]: @LinusU: Take your advice :)

**Review by tniessen — APPROVED** [review]

**Review by vsemozhetbyt — APPROVED** [review]
Doc format LGTM)

**Comment by ghost** [comment]
Thanks all!

**Pull request merged**